### PR TITLE
Add interface speed and more attributes

### PR DESCRIPTION
**Pull request by mingjunzhang2019**
SUMMARY
To support speed, auto-negotiate, advertised-speed and fec attributes by interface module

ISSUE TYPE
- Feature Pull Request

COMPONENT NAME
sonic_interfaces

ADDITIONAL INFORMATION
- regression test log: 
[regression_test.log](https://github.com/ansible-collections/dellemc.enterprise_sonic/files/10223030/regression_test.log)

- regression test result: 
[regression-2022-12-13-15-55-00.html.pdf](https://github.com/ansible-collections/dellemc.enterprise_sonic/files/10223039/regression-2022-12-13-15-55-00.html.pdf)

- CLI verification log: 
[cli_verification.log](https://github.com/ansible-collections/dellemc.enterprise_sonic/files/9872157/cli_verification.log)

```

### CODE REVIEW

**Review by kerry-meyer — COMMENTED** [review]
All of the structural changes look good.

I am posting some minor comments and questions on some details.

File: plugins/modules/sonic_interfaces.py [hunk 1]
```diff
@@ -63,6 +63,45 @@
...
+        description:
+        - Interface FEC (Forward Error Correction).
+        type: str
+        choices:
+        - FEC_RS
+        - FEC_FC
+        - FEC_DISABLED
+        - FEC_DEFAULT
+        - FEC_AUTO
   state:
```
[the remark this answers is not in the document]
Comment (mingjunzhang2019) [reply plugins/modules/sonic_interfaces.py:105]: Added replaced and overriden

File: tests/regression/roles/sonic_interfaces/defaults/main.yml [hunk 2]
```diff
@@ -54,6 +54,27 @@ tests:
         mtu: 5222
         enabled: true
   - name: test_case_03
+    description: Configure interface parameter speed
```
[the remark this answers is not in the document]
Comment (mingjunzhang2019) [reply tests/regression/roles/sonic_interfaces/defaults/main.yml:57]: SONIC on some platforms does not support some of them are configured together.

File: plugins/module_utils/network/sonic/config/interfaces/interfaces.py [hunk 3]
```diff
@@ -313,8 +331,12 @@ def is_this_delete_required(self, conf, have):
...
         intf = next((e_intf for e_intf in have if conf['name'] == e_intf['name']), None)
         if intf:
-            if (intf['name'].startswith('Loopback') or not ((intf.get('description') is None or intf.get('description') == '') and
-               (intf.get('enabled') is None or intf.get('enabled') is True) and (intf.get('mtu') is None or intf.get('mtu') == 9100))):
+            if (intf['name'].startswith('Loopback') or
+                not ((intf.get('description') is None or intf.get('description') == '') and
+                     (intf.get('enabled') is None or intf.get('enabled') is True) and
+                     (intf.get('mtu') is None or intf.get('mtu') == 9100) and
+                     (intf.get('fec') is None or intf.get('fec') == 'FEC_DISABLED') and
+                     (intf.get('auto_negotiate') is None or intf.get('auto_negotiate') is False))):
```
[the remark this answers is not in the document]
Comment (mingjunzhang2019) [reply plugins/module_utils/network/sonic/config/interfaces/interfaces.py:339]: They do not have common default.

File: plugins/module_utils/network/sonic/config/interfaces/interfaces.py [hunk 4]
```diff
@@ -335,20 +357,82 @@ def is_this_change_required(self, conf, have):
 
         return ret_flag
 
-    def build_create_payload(self, conf):
-        temp_conf = dict()
-        temp_conf['name'] = conf['name']
+    def build_create_config_request(self, conf):
```
[the remark this answers is not in the document]
Comment (mingjunzhang2019) [reply plugins/module_utils/network/sonic/config/interfaces/interfaces.py]: Changed to build_create_common_config_request.

File: plugins/modules/sonic_interfaces.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (kerry-meyer) [reply plugins/modules/sonic_interfaces.py:105]: Thank you.

File: plugins/module_utils/network/sonic/config/interfaces/interfaces.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (kerry-meyer) [reply plugins/module_utils/network/sonic/config/interfaces/interfaces.py:340]: It is okay to exclude the "or" condition in this block for attributes that don't have a common default.

(Cause the function to return True if either of these attributes is present in the "delete" request.)

[the remark this answers is not in the document]
Comment (mingjunzhang2019) [reply plugins/module_utils/network/sonic/config/interfaces/interfaces.py:340]: Discussed with Kerry, we got solution.

File: plugins/module_utils/network/sonic/config/interfaces/interfaces.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (kerry-meyer) [reply plugins/module_utils/network/sonic/config/interfaces/interfaces.py]: Thank you.

Resolved.

File: tests/regression/roles/sonic_interfaces/defaults/main.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (kerry-meyer) [reply tests/regression/roles/sonic_interfaces/defaults/main.yml:57]: Thanks for making updating the test case 4 description.

Resolved.

**Review by kerry-meyer — COMMENTED** [review]
I'm posting my "pending" comments, but I think we might have all of the issues resolved now. If so, I'll approve and merge this PR. (Otherwise, I will post another comment regarding any remaining issues.)

File: plugins/module_utils/network/sonic/config/interfaces/interfaces.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (kerry-meyer) [reply plugins/module_utils/network/sonic/config/interfaces/interfaces.py:340]: Please add one more line in this block to cover "advertised_speed" (below auto_negotiate?).

[the remark this answers is not in the document]
Comment (mingjunzhang2019) [reply plugins/module_utils/network/sonic/config/interfaces/interfaces.py]: Aded default interface speed handlin.

[the remark this answers is not in the document]
Comment (kerry-meyer) [reply plugins/module_utils/network/sonic/config/interfaces/interfaces.py]: The revisions and updated test results look good.

Thanks for making these additional changes.

I will "Resolve" this last remaining issue.

**Review by kerry-meyer — APPROVED** [review]
All review issues have been resolved.

The current change set looks good for merging.

Approved.

**Pull request merged**